test/intl402/Segmenter/constructor/constructor/options-granularity-word.js

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 Igalia, S.L. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-Intl.Segmenter
+description: Checks handling of valid values for the granularity option "word" when string text includes URLs, email addresses, #, Dates and acronyms
+features: [Intl.Segmenter]
+---*/
+
+const testValuesByLength = [
+  ["F.C.B", 1],
+  ["F.C.BALL", 1],
+  ["F.C. BALL", 4],
+  ["Please P.T.O 123 times. 123.3242432", 10],
+  ["at 12:43 PM", 7],
+  ["in 21/03/2013.", 8],
+  ["https://example.com/tc39/test262", 9],
+  ["https://example.com/tc39/test262?id=test", 13],
+  ["example.es , example.com they from era .com", 14],
+  ["124.13 1231,23", 3],
+  ["#test262, #TC39", 6],
+  ["my@mail.com sending message to my@mail.org", 13],
```

Comment:
Expecting a length of 13 based on JSC output but I believe the correct answer it's 9 due the segmentation of email addresses 

```
[Log] my
[Log] @
[Log] mail.com
[Log]  
[Log] sending
[Log]  
[Log] message
[Log]  
[Log] to
[Log]  
[Log] my
[Log] @
[Log] mail.org
```